lib/api_test.go

```diff
@@ -3,92 +3,409 @@ package sebak
 import (
 	"boscoin.io/sebak/lib/storage"
 	"bufio"
-	"context"
+	"bytes"
 	"encoding/json"
 	"fmt"
 	"github.com/gorilla/mux"
+	"github.com/stellar/go/keypair"
+	"github.com/stretchr/testify/assert"
+	"io/ioutil"
 	"net/http"
+	"net/http/httptest"
 	"sync"
 	"testing"
-	"time"
 )
 
-func TestStreaming(t *testing.T) {
+func prepareServer(pattern string, getHandlerFunc func(storage *sebakstorage.LevelDBBackend) http.HandlerFunc) (testServer *httptest.Server, storage *sebakstorage.LevelDBBackend, err error) {
+	// Setting Server
+	storage, err = sebakstorage.NewTestMemoryLevelDBBackend()
+	if err != nil {
+		return
+	}
+
+	router := mux.NewRouter()
+	router.HandleFunc(pattern, getHandlerFunc(storage)).Methods("GET")
+
+	testServer = httptest.NewServer(router)
+	return
+}
+
+func TestGetAccountHandler(t *testing.T) {
 
 	var wg sync.WaitGroup
-	wg.Add(1)
+	wg.Add(2)
 
 	// Setting Server
-	storageConfig, err := sebakstorage.NewConfigFromString("memory://")
-	if err != nil {
-		t.Errorf("failed to initialize file db: %v", err)
-
-	}
-	storage, err := sebakstorage.NewStorage(storageConfig)
-	if err != nil {
-		t.Errorf("failed to initialize file db: %v", err)
-	}
+	storage, _ := sebakstorage.NewTestMemoryLevelDBBackend()
 	defer storage.Close()
 
 	router := mux.NewRouter()
-	router.HandleFunc("/account/{address}", GetAccountHandler(storage)).Methods("GET")
-	server := &http.Server{Addr: ":5000", Handler: router}
-	go server.ListenAndServe()
+	router.HandleFunc(GetAccountHandlerPattern, GetAccountHandler(storage)).Methods("GET")
+
+	ts := httptest.NewServer(router)
+	defer ts.Close()
```

Comment:
In some circumstances, especially CircleCI or Travis, closing http server was hanged. We need stronger way to close server like `Shutdown`,
```
defer func() {
	timer := time.NewTimer(500 * time.Millisecond)
	ctx, cancel := context.WithCancel(context.Background())
	go func() {
		ts.Shutdown(ctx)
	}()
	<-timer.C
	cancel()
}()
```

Reply:
@kfangw Please see https://github.com/bosnet/sebak/pull/121#discussion_r198686738 .